lib/plugins/aws/deploy/lib/existsDeploymentBucket.js

```diff
@@ -23,6 +23,26 @@ module.exports = {
         return BbPromise.resolve();
       })
       .catch(err => {
+        if (err.message.indexOf('does not exist') > -1) { // create deployment bucket
+          const params = {
+            Bucket: bucketName,
+            CreateBucketConfiguration: {
+              LocationConstraint: this.provider.getRegion(),
+            },
+          };
+
+          this.serverless.cli.log(`Create deployment bucket ${bucketName}`);
+
+          return this.provider.request('S3',
+            'createBucket', params
+          ).then(() => BbPromise.resolve()
+          ).catch(error => {
+            throw new this.serverless.classes.Error(
+              `Can not create deployment bucket. Error: ${error.message}`
+            );
+          });
+        }
+
         throw new this.serverless.classes.Error(
```

Comment:
I don't see the need to throw this error since the bucket was either created or a error was thrown when trying to create it.

Reply:
Under what condition is that throw called? And is the message still accurate?